src/frontend/flight/sync/metrics.rs

```diff
@@ -0,0 +1,95 @@
+use metrics::{
+    counter, describe_counter, describe_gauge, describe_histogram, gauge, histogram,
+    Gauge, Histogram,
+};
+
+const REQUEST: &str = "seafowl_sync_writer_request";
+const IN_MEMORY: &str = "seafowl_sync_writer_in_memory";
+const COMPACTION_TIME: &str = "seafowl_sync_writer_compaction_time";
+const COMPACTED: &str = "seafowl_sync_writer_compacted";
+const FLUSHING_TIME: &str = "seafowl_sync_writer_flushing_time";
+const FLUSHED: &str = "seafowl_sync_writer_flushed";
```

Comment:
https://prometheus.io/docs/practices/naming/

The rule of thumb for these is that if you sum up some metric by all possible label values, it should still be a meaningful value, so we can't have different units in the same metric, instead:

```
# counter: total rows requested and total size of requests
seafowl_sync_writer_requests_rows_total
seafowl_sync_writer_requests_bytes_total

# gauges: how many rows in memory and how much space do they take
seafowl_sync_writer_in_memory_rows_total
seafowl_sync_writer_in_memory_bytes_total

# histogram: time taken by compaction
seafowl_sync_writer_compaction_time_seconds

# counter: total rows compacted, amount and size
seafowl_sync_writer_compacted_rows_total
seafowl_sync_writer_compacted_bytes_total

# histogram: time taken by flushes
seafowl_sync_writer_flushed_time_seconds

# counter: total rows flushed, amount and size
seafowl_sync_writer_flushed_bytes_total
seafowl_sync_writer_flushed_rows_total
```